userspace/libscap/examples/01-open/scap_open.c

```diff
@@ -20,42 +20,328 @@ limitations under the License.
-uint64_t g_nevts = 0;				   /* total number of events captured. */
-scap_t* g_h = NULL;				   /* global scap handler. */
-uint16_t* lens16 = NULL;			   /* pointer used to print the length of event params. */
-char* valptr = NULL;				   /* pointer used to print the value of event params. */
+scap_open_args oargs = {.engine_name = UNKNOWN_ENGINE};			    /* scap oargs used in `scap_open`. */
+uint64_t g_nevts = 0;							    /* total number of events captured. */
+scap_t* g_h = NULL;							    /* global scap handler. */
+uint16_t* lens16 = NULL;						    /* pointer used to print the length of event params. */
+char* valptr = NULL; /* pointer used to print the value of event params. */ /* pointer used to print the value of event params. */
+struct timeval tval_start, tval_end, tval_result;
+
+/*=============================== PRINT SUPPORTED SYSCALLS ===========================*/
+
+void print_sorted_syscalls(char string_vector[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_LEN], int dim)
+{
+	char temp[SYSCALL_NAME_MAX_LEN];
+
+	/* storing strings in the lexicographical order */
+	for(int i = 0; i < dim; ++i)
+	{
+		for(int j = i + 1; j < dim; ++j)
+		{
+			/* swapping strings if they are not in the lexicographical order */
+			if(strcmp(string_vector[i], string_vector[j]) > 0)
+			{
+				strcpy(temp, string_vector[i]);
+				strcpy(string_vector[i], string_vector[j]);
+				strcpy(string_vector[j], temp);
+			}
+		}
+	}
+
+	printf("\nSyscalls in the lexicographical order: \n");
+	for(int i = 0; i < dim; i++)
+	{
+		printf("[%d] %s\n", i, string_vector[i]);
+	}
+	printf("Interesting syscalls: %d\n", dim);
+}
+
+void print_UF_NEVER_DROP_syscalls()
+{
+	char str[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_LEN];
+	int interesting_syscall = 0;
+
+	for(int ppm_sc = 0; ppm_sc < PPM_SC_MAX; ppm_sc++)
+	{
+		for(int syscall_nr = 0; syscall_nr < SYSCALL_TABLE_SIZE; syscall_nr++)
+		{
+			if(g_syscall_code_routing_table[syscall_nr] != ppm_sc)
+			{
+				continue;
+			}
+
+			if(g_syscall_table[syscall_nr].flags & UF_NEVER_DROP)
+			{
+				strcpy(str[interesting_syscall++], g_syscall_info_table[ppm_sc].name);
+			}
+		}
+	}
+
+	printf("\n------- Print UF_NEVER_DROP syscalls: \n");
+	print_sorted_syscalls(str, interesting_syscall);
+}
+
+void print_EF_MODIFIES_STATE_syscalls()
+{
+	char str[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_LEN];
+	int interesting_syscall = 0;
+
+	for(int ppm_sc = 0; ppm_sc < PPM_SC_MAX; ppm_sc++)
+	{
+		for(int syscall_nr = 0; syscall_nr < SYSCALL_TABLE_SIZE; syscall_nr++)
+		{
+			if(g_syscall_code_routing_table[syscall_nr] != ppm_sc)
+			{
+				continue;
+			}
+
+			int enter_event = g_syscall_table[syscall_nr].enter_event_type;
+			if(g_event_info[enter_event].flags & EF_MODIFIES_STATE)
+			{
+				strcpy(str[interesting_syscall++], g_syscall_info_table[ppm_sc].name);
+			}
+		}
+	}
+
+	printf("\n------- Print EF_MODIFIES_STATE syscalls: \n");
+	print_sorted_syscalls(str, interesting_syscall);
+}
+
+void print_both_syscalls()
```

Comment:
This function name is not very clear, any way we can rename it to something more meaningful?